### PR TITLE
Portals 3311 - EL Portal Contribute Your Data Component

### PR DESCRIPTION
**Pull request by afwillia**
This PR addresses [Portals-3311](https://sagebionetworks.jira.com/browse/PORTALS-3311) by adding a static component to the EL Portal "Contribute Data" tab. The design is from the [EL Portal Redesign in Figma](https://www.figma.com/design/Xbf2MAuWJjdntMbb2wKSD5/ELITE-Portal-Redesign?node-id=104-9801&node-type=canvas&t=amC47ucClglElSNn-0).

- [apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx](https://github.com/Sage-Bionetworks/synapse-web-monorepo/compare/main...afwillia:PORTALS-3311?expand=1#diff-be722161b4cf96da51ed892272823eeca5b91ef4f1d246a103a89077255b1742) is a react component with two background images, `dot_blob_top_left.png` and `dot_blob_bottom_right.png`, and explanatory text with a link to file a data contribution ticket.

I have some general questions for reviewers about testing, project structure, and accurately reproducing Figma designs. Here are some specifics for this PR:

- Have I added the types correctly to `synapseConfigArray`? 
- I want to add tests, what's the best way to add unit tests for this component?
- This component contains hard-coded styling; is there a better way to add this?
- The current output does not completely match the styling of the Figma design. Do you have any tips for getting the padding, spacing, bars, general alignment, etc. to match the design?
- Any other questions, comments, suggestions!

### CODE REVIEW

**Comment by afwillia** [comment]
[This workflow failed](https://github.com/Sage-Bionetworks/synapse-web-monorepo/actions/runs/11730390315/job/32678646361#step:4:404) `elportal:typecheck` because 

>  Error: src/config/routesConfig.tsx(404,13): error TS2322: Type '"ELContributeYourData"' is not assignable to type '"RedirectToURL" | "RedirectWithQuery" | "Redirect" | "RouteControlWrapper" | "CardContainerLogic" | "QueryWrapper" | "UserCard" | "Markdown" | "MarkdownCollapse" | "DetailsPage" | ... 48 more ... | "DynamicFormModal"'.

**Comment by afwillia** [comment]
tagging @jay-hodgson for review, thanks!

File: apps/portals/elportal/src/config/routesConfig.tsx
Comment (nickgros) [inline apps/portals/elportal/src/config/routesConfig.tsx:405]: This will remove the undesirable side padding
```suggestion
            props: undefined,
            isOutsideContainer: true
```

This boolean determines whether the component will be wrapped in a container. Note that I'm trying to get rid of this with the routing overhaul, so in the future you would simply wrap (or not wrap) the `<ELContributeYourData />` JSX in a `<Container>` element

File: apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx
Comment (nickgros) [inline apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx:53]: You can use the MUI Typography component here, and IMO we should update the font in the theme used in the portal instead of here.

```suggestion
        <Typography variant="headline1" style={{ color: 'white' }}>
          Contribute Your Data
        </Typography>
```

File: apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx
Comment (nickgros) [inline apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx:58]: Typography again, for consistency
```suggestion
        <Typography variant="body1" style={{ color: 'white' }}>
          If you are a funded portal contributor and ready to upload data to the
          ELITE Portal, you can begin the data submission process by contacting
          our data curation team through our service desk.
        </Typography>
```

File: apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx
Comment (nickgros) [inline apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx:71]: For this custom hover color, slightly modify the base color so it feels 'interactive'. You can import `darken` from @material/mui - feel free to tweak the second argument as desired
```suggestion
                backgroundColor: darken('#5BA998', 0.05),
```

File: apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx
Comment (nickgros) [inline apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx:50]: Since this horizontal line seems to be the same width as the headline text, I would suggest making it a `borderTop` property on the headline component, maybe adding these `sx` properties to the `Typography` component would work?

```
            pt: 2,
            mx: 'auto',
            width: 'max-content',
            borderTop: '6px solid #ffffff88',
```

**Review by nickgros — APPROVED** [review]
>  Have I added the types correctly to synapseConfigArray?

Yes! Note I'm planning to remove this API in #1276 in favor of just writing JSX

>  I want to add tests, what's the best way to add unit tests for this component?

You could add a co-located `ELContributeYourData.test.tsx` file and use react-testing-library to render and inspect your component. We unfortunately have poor test quality in the `synapse-portal-framework`, but we have a lot of examples in `synapse-react-client`. `AccountLevelBadge.test.tsx` might be a decent example to look at.

> This component contains hard-coded styling; is there a better way to add this?

In general, we can try to rely on the MUI theme. I think what you did doesn't diverge from our current practices much. I think it might be a good idea if for the team to consider adopting something like CSS modules when we don't need CSS-in-JS (like this case). But as of now we haven't done that.

> The current output does not completely match the styling of the Figma design. Do you have any tips for getting the padding, spacing, bars, general alignment, etc. to match the design?

This is a lot of tricky CSS, so this was a good first pass! I added comments and suggestions in the PR

File: apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx
Comment (afwillia) [reply apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx:50]: thank you! That was bugging me, I couldn't get the bar to work fit `fit-content`. I'll give this a shot.

File: apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx
Comment (nickgros) [inline apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx:43]:
```suggestion
            pt: 4,
            mb: 2,
```

File: apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx
Comment (nickgros) [inline apps/synapse-portal-framework/src/components/elportal/ELContributeYourData.tsx:51]:
```suggestion
        <Typography
          variant="body1"
          sx={{ color: 'white', fontStyle: 'italic', mb: 3 }}
        >
```

**Review by nickgros — APPROVED** [review]
Couple of small tweaks to get it closer to the design. IMO good to merge if you're ok with these last suggested changes! Tests look good!

**Comment by afwillia** [comment]
@nickgros those changes look good to me, thanks! I just marked the PR as ready and will merge once the workflows run.

**Comment by afwillia** [comment]
@nickgros looks like I don't have write access to merge this PR. Can you merge it or give me access?

**Comment by nickgros** [comment]
I think @jay-hodgson would have to add you to the [Web Engineering team](https://github.com/orgs/Sage-Bionetworks/teams/web-engineering)

**Pull request merged**
